src/ClientHandle.cpp

```diff
@@ -1591,7 +1619,26 @@ void cClientHandle::HandleUpdateSign(
 void cClientHandle::HandleUseEntity(UInt32 a_TargetEntityID, bool a_IsLeftClick)
 {
 	// TODO: Let plugins interfere via a hook
-	
+
+	// If the player is a spectator, let him spectate
+	if (m_Player->IsGameModeSpectator() && a_IsLeftClick)
+	{
+		class cSpectateEntity : public cEntityCallback
+		{
+			cPlayer & m_Player;
+			bool Item(cEntity * a_Entity) override
+			{
+				m_Player.AttachTo(a_Entity);
+				return true;
+			}
+		public:
+			cSpectateEntity(cPlayer & a_Player) : m_Player(a_Player) {}
+		} Callback(*m_Player);
+
+		m_Player->GetWorld()->DoWithEntityByID(a_TargetEntityID, Callback);
```

Comment:
I've always seen this kind of Callback-thing. Should I only change my callbacks or any other callbacks?